config/traefik.yaml

```diff
@@ -0,0 +1,113 @@
+entrypoints:
+  # Defines a secure entrypoint using TLS encryption
+  websecure:
+    address: ":443"
+    http:
+      tls: true
+  # Defines an insecure entrypoint that redirects to the secure one.
+  web:
+    address: ":80"
+    http:
+      # Redirects HTTP to HTTPS
+      redirections:
+        entrypoint:
+          to: "websecure"
+          scheme: "https"
+
+api:
+  insecure: true
+  dashboard: true
+
+http:
+  services:
+    grist:
+      loadBalancer:
+        servers:
+          - url: 'http://grist:8484'
+    dex:
+      loadBalancer:
+        servers:
+          - url: 'http://dex:5556'
+    authelia:
+      loadBalancer:
+        servers:
+          - url: 'http://authelia:9091'
+
+  middlewares:
+    no-fwd:
+      headers:
+        customRequestHeaders:
+          'X-Forwarded-User': ""
+
+  routers:
+    route-grist-login:
+      rule: "PathPrefix(`/auth/login`) || PathPrefix(`/_oauth`)"
+      service: grist
+      entryPoints:
+        - web
+
+    route-grist:
+      rule: "PathPrefix(`/`)"
+      priority: 1   # Set a lower priority than the other rules
+      service: grist
+      middlewares:
+        - no-fwd
+      entryPoints:
+        - web
+
+    route-dex:
+      rule: "PathPrefix(`/dex/`) || Path(`/dex`)"
+      service: dex
+      entryPoints:
+        - web
+
+    route-authelia:
+      rule: "PathPrefix(`/authelia/`) || Path(`/authelia`)"
+      service: authelia
+      entryPoints:
+        - web
+
+{{ $use_https := env "USE_HTTPS" }}
+{{if eq $use_https "true" }}
+    https-route-grist-login:
+      rule: 'Host(`{{ env "APP_HOST" }}`) && (PathPrefix(`/auth/login`) || PathPrefix(`/_oauth`))'
+      service: grist
+      entryPoints:
+        - websecure
+      tls: {{ env "TLS" }}
```

Comment:
Does this mean the "TLS" variable needs to be valid YAML? 

The Traefik docs suggest this should be `{}`, but I'm not familiar enough with Traefik to know what happens when you put in an env variable (like this).

Reply:
This is a vestige from Omnibus. Yes, it's supposed to expand to valid YAML.

I need to do more realistic HTTPS testing to see if I can make this a smooth experience for the user, at least better than the existing one in AWS.

Reply:
So I've tested this and this works. I've changed this to use somewhat different templating but it works with both self-signed certs and Let's Encrypt.